### PR TITLE
Removal "cache-and-network" strategy from multiple queries

### LINKED ISSUE #412: Minimize cache-and-network strategies
### Description

After all the UX changes are implemented, we can now revisit the `cache-and-network` strategies for our queries. Now that we utilize caches a lot, we are able to remove some redundant requests that are marked as cache-and-network  

### Acceptance Criteria

- Alter some of our `cache-and-network` strategies to the default `cache-first` strategy

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

After completing all other issues in #407, this is the last task for optimising queries and cached data.
This includes updating multiple queries from `fetchPolicy: "cache-and-network"` to default Apollo policy  `cache-first`.

This change skips unnecessary fetching queries for data we already have and instead displays data from cache. 

Unchanged queries on components:
- `ListAlerts`
- `AlertDetails`
- `ComplianceOverview`
- `ListPolicies`
- `ListRules`
- `PolicyDetails`
- `ResourceDetails`
- `RuleDetails`
- `LIstResources`

Closes #412 
## Changes

Changed queries on components: 
- `Destinations`
- `EditPolicy`
- `EditRule`
- `ListUsers`

## Testing

- Locally

### CODE REVIEW

File: web/src/pages/ListResources/ListResources.tsx [hunk 1]
```diff
@@ -44,7 +44,6 @@ const ListResources = () => {
   } = useRequestParamsWithPagination<ListResourcesInput>();
 
   const { loading, data, error } = useListResources({
-    fetchPolicy: 'cache-and-network',
```
Comment (3nvi) [inline web/src/pages/ListResources/ListResources.tsx:47]: that should stay here. A resource can go from PASS to FAIL in the background without a user action

File: web/src/pages/PolicyDetails/PolicyDetails.tsx [hunk 2]
```diff
@@ -53,7 +53,6 @@ const PolicyDetailsPage = () => {
   >();
 
   const { error, data, loading } = usePolicyDetails({
-    fetchPolicy: 'cache-and-network',
```
Comment (3nvi) [inline web/src/pages/PolicyDetails/PolicyDetails.tsx:56]: same as above

File: web/src/pages/ResourceDetails/ResourceDetails.tsx [hunk 3]
```diff
@@ -54,7 +54,6 @@ const ResourceDetailsPage = () => {
   >();
 
   const { error, data, loading } = useResourceDetails({
-    fetchPolicy: 'cache-and-network',
```
Comment (3nvi) [inline web/src/pages/ResourceDetails/ResourceDetails.tsx:57]: same as above

File: web/src/pages/RuleDetails/RuleDetails.tsx [hunk 4]
```diff
@@ -31,7 +31,6 @@ import RuleDetailsAlertsTable from './RuleDetailsAlertsTable';
 const RuleDetailsPage = () => {
   const { match } = useRouter<{ id: string }>();
   const { error, data, loading } = useRuleDetails({
-    fetchPolicy: 'cache-and-network',
```
Comment (3nvi) [inline web/src/pages/RuleDetails/RuleDetails.tsx:34]: same as above

**Comment by alexmylonas** [comment]
@3nvi Reverted policies based on your suggestions!

**Pull request merged**